server/welcome.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+type welcomeFeedbackMigration struct {
+	CreateAt time.Time
+}
+
+// setWelcomeFeedbackMigration in called on plugin activation to set the time when the welcome feedback
+// has been enabled on this server, preventing us to send the welcome feedback to older users
+func (p *Plugin) setWelcomeFeedbackMigration(now time.Time) {
+	var migration welcomeFeedbackMigration
+	// set the current date in the KV store if it does not exist.
+	if err := p.KVGet(WelcomeFeedbackMigrationKey, &migration); err != nil {
+		p.API.LogError("Failed to get welcome feedback migration key", "err", err)
+		return
+	}
+
+	if migration.CreateAt.IsZero() {
+		migration = welcomeFeedbackMigration{CreateAt: now}
+		p.API.LogInfo("Setting welcome feedback migration date", "date", now.String())
+		if err := p.KVSet(WelcomeFeedbackMigrationKey, migration); err != nil {
+			p.API.LogError("Failed to set welcome feedback migration key", "err", err)
+			return
+		}
+	}
+
+	p.welcomeFeedbackAfter = migration.CreateAt.Add(-TimeUntilWelcomeFeedback)
```

Comment:
Yes, anyone having an account created after `"the date when the plugin was installed" - 7days` will get the question. So we are covering as many people as possible ~ slightly overkill, but this would allow us to get data "sooner"